### PR TITLE
[week5] - sungmin

### PR DESCRIPTION
**Pull request by csm3310**
Complete week5 assignment

### CODE REVIEW

**Review by giyomdabin — APPROVED** [review]
task2:  주어진 과제를 잘 수행하셨습니다.

코드의 완성도를 높이기 위해 다음 방법을 고려할 수 있습니다:

- 드롭아웃 레이어 추가: 과적합 방지를 위해 드롭아웃 레이어를 추가하면 모델의 일반화 성능을 강화할 수 있습니다.
- MSE 외에 MAE(Mean Absolute Error)를 추가로 계산하면 모델 성능을 다각도로 평가할 수 있습니다.

MSE는 **큰 오차** 에 민감한 지표로, MSE만을 기준으로 모델 성능을 평가하면, 극단적인 오차에 영향을 많이 받아 모델의 평균적인 성능을 과소평가할 가능성이 있습니다.
MAE는 큰 오차에 민감하지 않으며 모든 데이터에 걸쳐 평균적인 오차 크기를 직접적으로 나타냅니다.
따라서 두 지표를 함께 분석하면 모델 성능을 더 다각적으로 평가할 수 있으며, 특정 데이터의 특성에 맞게 개선 방향을 도출할 수 있습니다.

수고하셨습니다.

**Comment by kimhamyong** [comment]
task1:

초기 학습 손실이 비정상적으로 클 경우 그래디언트 폭주를 방지하기 위해 Gradient Clipping을 적용하는 방법이 있습니다.

```python
transfer_model.compile(
    optimizer=tf.keras.optimizers.Adam(learning_rate=0.001, clipnorm=1.0),
    loss=tf.keras.losses.SparseCategoricalCrossentropy(),
    metrics=['accuracy']
)
```

**Comment by kimhamyong** [comment]
task1:

발생한 경고메시지 해결법에 대해 알려드립니다.
WARNING:absl:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.

- load_model()로 모델을 불러오면 저장 당시의 옵티마이저, 손실 함수, 메트릭 정보도 함께 복원됩니다. 그러나 모델의 구조가 변경되거나 새로운 레이어가 추가된 경우, 기존 컴파일 정보가 더 이상 유효하지 않게 됩니다. 이런 상태에서 평가나 학습을 시도하면 경고 메시지가 발생하게 됩니다.

- 1. compile()을 호출하여 새로운 옵티마이저, 손실 함수, 메트릭을 명시적으로 정의하는 방법이 있습니다.
```python
base_model = load_model('my_model.h5')
base_model.compile(
    optimizer=tf.keras.optimizers.Adam(),
    loss=tf.keras.losses.SparseCategoricalCrossentropy(),
    metrics=['accuracy']
)
```

- 2. load_model() 호출 시 compile=False 옵션을 설정하여 저장된 컴파일 정보를 무시하는 방법이 있습니다.
```python
base_model = load_model('my_model.h5', compile=False)
```

**Review by kimhamyong — APPROVED** [review]
task1: 잘 수행하셨습니다.

마지막 과제까지 수고 많으셨습니다!
